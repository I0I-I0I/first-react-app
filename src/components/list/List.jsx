import Item from "./item/Item";

const List = ({ onRemoveItem, list }) => (
	<ul>
		{list.map((item) => (
			<Item key={item.objectIDkey} item={item} remove={onRemoveItem} />
		))}
	</ul>
);

export default List;
