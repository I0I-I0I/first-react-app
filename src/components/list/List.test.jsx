import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import List from "./List";

const storyOne = {
	title: "React",
	url: "https://reactjs.org/",
	author: "Jordan Walke",
	num_comments: 3,
	points: 4,
	objectID: 0,
};

const storyTwo = {
	title: "Redux",
	url: "https://redux.js.org/",
	author: "Dan Abramov, Andrew Clark",
	num_comments: 2,
	points: 5,
	objectID: 1,
};

const stories = [storyOne, storyTwo];

describe("List", () => {
	const onRemoveItem = vi.fn();

	it("count items after rendering", () => {
		render(<List list={stories} onRemoveItem={onRemoveItem} />);
		const list = screen.getAllByRole("listitem");
		expect(list.length).toEqual(2);
	});

	it("clicking the dismiss button calls the callback handler", () => {
		render(<List list={stories} onRemoveItem={onRemoveItem} />);
		const button = screen.getAllByRole("button")[0];
		fireEvent.click(button);
		expect(onRemoveItem).toHaveBeenCalledTimes(1);
	});

	it("renders snapshot", () => {
		const { container } = render(
			<List list={stories} onRemoveItem={onRemoveItem} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
