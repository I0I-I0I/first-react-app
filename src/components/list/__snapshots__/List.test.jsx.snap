// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List > renders snapshot 1`] = `
<ul>
  <li
    class="_item_d7d403"
  >
    <a
      href="https://reactjs.org/"
      style="width: 40%;"
      target="_blank"
    >
      React
    </a>
    <span
      style="width: 30%;"
    >
      Jordan Walke
    </span>
    <span
      style="width: 10%;"
    >
      3
    </span>
    <span
      style="width: 10%;"
    >
      4
    </span>
    <span
      style="width: 10%;"
    >
      <button
        class="_button_3c5126 _buttonSmall_3c5126"
      >
        delete
      </button>
    </span>
  </li>
  <li
    class="_item_d7d403"
  >
    <a
      href="https://redux.js.org/"
      style="width: 40%;"
      target="_blank"
    >
      Redux
    </a>
    <span
      style="width: 30%;"
    >
      Dan Abramov, Andrew Clark
    </span>
    <span
      style="width: 10%;"
    >
      2
    </span>
    <span
      style="width: 10%;"
    >
      5
    </span>
    <span
      style="width: 10%;"
    >
      <button
        class="_button_3c5126 _buttonSmall_3c5126"
      >
        delete
      </button>
    </span>
  </li>
</ul>
`;
